docs/getting-started.md

```diff
@@ -0,0 +1,152 @@
+
+- Install minishift using this [installation guide](https://docs.openshift.org/latest/minishift/getting-started/installing.html)
+- Install ocdev using this [installation guide](/README.md#Installation)
+
+Next, start a local OpenShift cluster using minishift -
+```console
+$ minishift start       
+-- Starting profile 'minishift'                                
+-- Checking if requested hypervisor 'kvm' is supported on this platform ... OK
+-- Checking if KVM driver is installed ...                     
+   Driver is available at /usr/local/bin/docker-machine-driver-kvm ... 
+   Checking driver binary is executable ... OK                 
+-- Checking if Libvirt is installed ... OK                     
+-- Checking if Libvirt default network is present ... OK       
+-- Checking if Libvirt default network is active ... OK        
+-- Checking the ISO URL ... OK 
+-- Starting local OpenShift cluster using 'kvm' hypervisor ... 
+-- Starting Minishift VM ............... OK                    
+-- Checking for IP address ... OK                              
+-- Checking if external host is reachable from the Minishift VM ... 
+   Pinging 8.8.8.8 ... OK      
+-- Checking HTTP connectivity from the VM ...                  
+   Retrieving http://minishift.io/index.html ... OK            
+-- Checking if persistent storage volume is mounted ... OK     
+-- Checking available disk space ... 19% used OK               
+-- OpenShift cluster will be configured with ...               
+   Version: v3.7.1             
+-- Checking 'oc' support for startup flags ...                 
+   routing-suffix ... OK       
+   host-config-dir ... OK      
+   host-data-dir ... OK        
+   host-pv-dir ... OK          
+   host-volumes-dir ... OK     
+Starting OpenShift using openshift/origin:v3.7.1 ...           
+OpenShift server started.
+
+The server is accessible via web console at:
+    https://192.168.42.147:8443
+```
+
+Now login to the OpenShift cluster using -
+```console
+$ oc login -u developer -p developer https://192.168.42.147:8443
+Login successful.
+
+You have one project on this server: "myproject"
+
+Using project "myproject".
+```
+
+Make sure you are logged in to the cluster by running `oc whoami` command.
+
+Now we can move on to creating our application using `ocdev`.
+
+#### Create an application
+
+An application is the umbrella under which you will build all the components (microservices) of your entire application.
+
+Let's create an application -
+
+```console
+$ ocdev application create nodeapp 
+Creating application: nodeapp
+Switched to application: nodeapp
+```
+
+#### Create a component
+
+Now that you have created an application, now add a component of type _nodejs_ to the application, from the current directory where our code lies.
```

Comment:
hmm, then I think we should have `--dir` in `ocdev component create` because it does not make any sense for the end user to create an empty application

Reply:
Why not? The component can be created beforehand, and then you dojust `ocdev push`. This is how most of the use cases are constructed.

It is nice and straightforward:
```
ocdev create nodejs
ocdev push
# edit
ocdev push
# edit
ocdev push
...
...
```

Reply:
I thought that `ocdev component create nodejs` would also push the current directory, which it does not.
As a developer, when I do `ocdev component create nodejs`, it's very weird to not have my code in it. Also, this will break when we don't do only the predefined templates and image streams that we're doing currently. Why would I want to deploy a basic template of a nodejs application without my code in it? It's a waste deployment and wastage of resources imo :|
I'd rather push the use case where I specify my current directory using `--dir .` so that at least I'm deploying my code, and not some barebones application that has nothing to do with my code.

Reply:
> I thought that ocdev component create nodejs would also push the current directory, which it does not.

Yeh, you might be right. We have look at the use cases and find out what should be correct behavior.

> I'd rather push the use case where I specify my current directory using --dir . so that at least I'm deploying my code, and not some barebones application that has nothing to do with my code.
> 

ok lets do it like you are suggesting